tests/test_font.py

```diff
@@ -11,402 +11,265 @@
-        self.assertFont(
-            Font("Comic Sans", 12, weight=BOLD), "Comic Sans", 12, NORMAL, NORMAL, BOLD
-        )
-
-        self.assertFont(
-            Font("Comic Sans", 12, weight="bold"),
-            "Comic Sans",
-            12,
-            NORMAL,
-            NORMAL,
-            BOLD,
-        )
-
-        self.assertFont(
-            Font("Comic Sans", 12, weight="something else"),
-            "Comic Sans",
+
+def test_equality():
+    assert Font("Comic Sans", "12 pt") == Font("Comic Sans", 12, NORMAL, NORMAL, NORMAL)
+
+
+def test_hash():
+    assert hash(Font("Comic Sans", 12)) == hash(Font("Comic Sans", 12))
+
+    assert hash(Font("Comic Sans", 12, weight=BOLD)) != hash(Font("Comic Sans", 12))
+
+
+@pytest.mark.parametrize(
+    "size, kwargs, string",
+    [
+        (12, {}, "12pt"),
+        (12, {"style": ITALIC}, "italic 12pt"),
+        (12, {"style": ITALIC, "variant": SMALL_CAPS}, "italic small-caps 12pt"),
+        (
             12,
-            NORMAL,
-            NORMAL,
-            NORMAL,
-        )
+            {"style": ITALIC, "variant": SMALL_CAPS, "weight": BOLD},
+            "italic small-caps bold 12pt",
+        ),
+        (12, {"variant": SMALL_CAPS, "weight": BOLD}, "small-caps bold 12pt"),
+        (12, {"weight": BOLD}, "bold 12pt"),
+        (12, {"style": ITALIC, "weight": BOLD}, "italic bold 12pt"),
+        # Check system default size handling
+        (SYSTEM_DEFAULT_FONT_SIZE, {}, "system default size"),
+        (SYSTEM_DEFAULT_FONT_SIZE, {"style": ITALIC}, "italic system default size"),
+    ],
+)
+def test_repr(size, kwargs, string):
+    assert repr(Font("Comic Sans", size, **kwargs)) == f"<Font: {string} Comic Sans>"
 
-    def test_make_normal_weight(self):
-        f = Font("Comic Sans", 12)
-        self.assertFont(f.normal_weight(), "Comic Sans", 12, NORMAL, NORMAL, NORMAL)
 
-        f = Font("Comic Sans", 12, weight=BOLD)
-        self.assertFont(f.normal_weight(), "Comic Sans", 12, NORMAL, NORMAL, NORMAL)
+@pytest.mark.parametrize("size", [12, "12", "12pt", "12 pt"])
+def test_simple_construction(size):
+    assert_font(Font("Comic Sans", size), "Comic Sans", 12, NORMAL, NORMAL, NORMAL)
 
-    def test_make_bold(self):
-        f = Font("Comic Sans", 12)
-        self.assertFont(f.bold(), "Comic Sans", 12, NORMAL, NORMAL, BOLD)
 
+def test_invalid_construction():
+    with pytest.raises(ValueError):
+        Font("Comic Sans", "12 quatloos")
 
-class ParseFontTests(TestCase):
-    def assertFont(self, font, family, size, style, variant, weight):
-        self.assertEqual(font.family, family)
-        self.assertEqual(font.size, size)
-        self.assertEqual(font.style, style)
-        self.assertEqual(font.variant, variant)
-        self.assertEqual(font.weight, weight)
 
-    def test_font_instance(self):
-        f = Font("Comic Sans", 12)
+@pytest.mark.parametrize(
+    "family",
+    ["Comics Sans", "Wingdings", "'Comic Sans'", '"Comic Sans"'],
+)
```

Comment:
Even when all the options for on a single line, style preference is to put each option on a line by itself, so that it's easier to parse the test variants that will be run.
```suggestion
@pytest.mark.parametrize(
    "family",
    [
        "Comics Sans", 
        "Wingdings", 
        "'Comic Sans'", 
        '"Comic Sans"',
    ],
)
```
This is especially true when there's more than one parameterised argument; the only real exception is when the parameter values are absurdly short (e.g., True/False, or the font size example above where the entire parameterization fits on one line)